ZeldaFullEditor/ZSCustomOverworld.asm

```diff
@@ -2835,33 +3134,72 @@ ReplaceBGColor:
 {
     PHB : PHK : PLB
 
-    ; TODO: This may need to check if we are in a warp and then if so load the
-    ; custom color. If not, then chceck if its enabled or not.
-    ;SEP #$20 ; Set A in 8bit mode.
+    SEP #$20 ; Set A in 8bit mode.
 
-    ;LDA.w Pool_EnableBGColor : BNE .custom
-        ;REP #$20 ; Set A in 16bit mode.
+    LDA.w Pool_EnableBGColor : BNE .custom
+        REP #$20 ; Set A in 16bit mode.
 
-        ;PLB
+        PLB
 
-        ;RTL
+        RTL
 
-    ;.custom
+    .custom
 
-    ;REP #$20 ; Set A in 16bit mode.
+    REP #$20 ; Set A in 16bit mode.
 
-    LDA.b $8A : ASL : TAX ; Get area code and times it by 2.
-    LDA.w Pool_BGColorTable, X ; Get the color.
+    ; TODO: Saving the Y may not be necessary, done just in case.
+    PHY
 
-    ;STA.l $7EC300 : STA.l $7EC340 ; Set the BG color.
+    ; Get area code and times it by 2.
+    LDA.b $8A : ASL : TAX
+
+    ; Get the color.
+    LDA.w Pool_BGColorTable, X
+
+    ; Set the BG color.
     STA.l $7EC500 : STA.l $7EC540
 
+    TAY ; Save the color.
+    
+    SEP #$20 ; Set A in 8bit mode.
+
+    ; TODO: Pretty sure this is needed. Just keep an eye out for it.
+    ; Set the buffer color when exiting to the OW to prevent a bug when using the
+    ; map in an area with a subscreen overlay.
+    LDA.b $10 : CMP.b #$08 : BNE .notPreOverworld
+        BRA .setBuffer
+
+    .notPreOverworld
+
+    ; TODO: Check if this is needed. I think it is. If not, it should always
+    ; set the buffer too. If not the warp fades into the wrong color for a
+    ; second.
+    ; Only set the buffer color during warps.
+    LDA.b $11 : CMP.b #$23 : BNE .notWarp
+        .setBuffer
+
+        REP #$20 ; Set A in 16bit mode.
+
+        TYA
+
+        ; Set the BG color buffer.
+        STA.l $7EC300 : STA.l $7EC340
+
+    .notWarp
+
+    REP #$20 ; Set A in 16bit mode.
+
+    PLY
+
     PLB
 
     RTL
 }
 
 ; TODO: Doccument this better and fiture out what it actually does.
+; This sets the initial scroll offsets for the pyramid BG. It seems highly
```

Comment:
It looks like it's making the offsets based on the position of the other background (BG1H vs BG2H on line 3212). Educated guess is maybe a slight parallax effect on the one axis or maybe just the boundary at which it stops moving the background layer. Can probably remove the TODO since we have done some triaging, but name could also be improved. SpecialBgHorizOffsetAdjustment or something to that effect

Reply:
Done.